src/ddmd/expressionsem.d

```diff
@@ -8771,6 +8784,101 @@ Expression binSemanticProp(BinExp e, Scope* sc)
     return null;
 }
 
+/********************************************************************************
+ * Helper function to resolve `@property` functions in a `BinAssignExp`.
+ * It rewrites expressions of the form, for example, `s.x += 1` to `s.x(s.x() += 1)`
+ * if `s` has a both a getter and setter `@property` named `x`.
+ * Params:
+ *      e = the expression to rewrite
+ *      sc = the semantic scope
+ * Returns:
+ *      the rewritten expression if the procedure succeeds, an `ErrorExp` if the
+ *      and error is encountered, or `null` if `e.e1` is not an `@property` function.
+ */
+Expression binSemanticProp(BinAssignExp e, Scope* sc)
+{
+    if (Expression ex = binSemantic(e, sc))
+        return ex;
+    Expression e1x = resolveProperties(sc, e.e1);
+    Expression e2x = resolveProperties(sc, e.e2);
+    if (e1x.op == TOKerror)
+        return e1x;
+    if (e2x.op == TOKerror)
+        return e2x;
+
+    //********************************************************************
+    // Check for property assignment.
+    // https://issues.dlang.org/show_bug.cgi?id=8006
+    if (e1x.op == TOKcall)
+    {
+        CallExp ce = cast(CallExp)e1x;
+
+        // Consider s.x() += 1;
+        // We want to rewrite that expression to s.x(s.x() + 1);
+
+        // Only rewrite @property functions.  Don't rewrite @ref @property functions.
```

Comment:
If there's a getter annotated with `ref` and a setter. If there's no rewrite for `ref` functions the setter will be bypassed.